File: app/src/main/java/com/example/bykov/roomdbapp/local/UserDAO.java
package com.example.bykov.roomdbapp.local;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import com.example.bykov.roomdbapp.model.User;

import java.util.List;

import io.reactivex.Flowable;

/**
 * Created by bykov on 13.06.2018.
 */

@Dao
public interface UserDAO {
    @Query("SELECT * FROM users WHERE id=:userId")
    Flowable<User> getUserById(int userId);

    @Query("SELECT * FROM users")
    Flowable<List<User>> getAllUsers();

    @Insert
    void insertUser(User... users);

    @Update
    void updateUser(User... users);

    @Delete
    void deleteUser(User user);

    @Query("DELETE FROM users")
    void deleteAllUsers();
}

File: app/src/main/java/com/example/bykov/roomdbapp/local/UserDAO.kt
package com.example.bykov.roomdbapp.local

import android.arch.persistence.room.*
import com.example.bykov.roomdbapp.model.User
import io.reactivex.Flowable

/**
 * Created by bykov on 13.06.2018.
 */
@Dao
interface UserDAO {
    @Query("SELECT * FROM users WHERE id=: userId")
    fun getUserById(userId: Int): Flowable<User>

    @Query("SELECT * FROM users")
    fun getAllUsers(): Flowable<List<User>>

    @Insert
    fun insertUser(user: User)

    @Update
    fun updateUser(user: User)

    @Delete
    fun deleteUser(user: User)

    @Query("DELETE FROM users")
    fun deleteAllUsers()
}